qt_ui/windows/mission/flight/QFlightCreator.py

```diff
@@ -103,18 +103,22 @@ def __init__(self, game: Game, package: Package, parent=None) -> None:
         # When an off-map spawn overrides the start type to in-flight, we save
         # the selected type into this value. If a non-off-map spawn is selected
         # we restore the previous choice.
-        self.restore_start_type: Optional[str] = None
+        self.restore_start_type = self.game.settings.default_start_type
```

Comment:
Initializing `self.restore_start_type` fixes an issue where you try to create a flight, if it happens to select a squadron with an "OffMapSpawn" base as a first choice, when you change to a squadron "on map" it'll enable the combo-box but it'll still be "in-flight". This fix makes sure it goes back to default start type, subsequent changes ought to behave the same...

Reply:
Extracted to its own PR...